modules/01 JavaScript Programming/Assignments/30/answer30.txt

```diff
@@ -0,0 +1,39 @@
+function ShapeType (name1,desc)
+{
+    this.name=name1;
+    this.description=desc;
+}
+
+function Shape (shapeType,width, height)
+{
+    this.shapeType=new ShapeType(shapeType.name,shapeType.description);
```

Comment:
you should've not create a new instance of "ShapeType". You already have an instance as an argument "shapeType"